src/SonarScanner.MSBuild.Shim/SonarScanner.Wrapper.cs

```diff
@@ -96,7 +97,14 @@ private static bool InternalExecute(AnalysisConfig config, IEnumerable<string> u
         private static string FindScannerExe()
         {
             var binFolder = Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location);
-            var fileExtension = PlatformHelper.IsWindows() ? ".bat" : "";
+            var scannerCliFolder = Path.Combine(binFolder, $"sonar-scanner-{SonarScannerVersion}");
+            if (!Directory.Exists(scannerCliFolder))
```

Comment:
note: given that the .NET Framework packaging won't do this, this code can be conditionally compiled
_(with a comment leading to the package-artifacts.ps1 script to link to the packaging)_

Reply:
@mary-georgiou-sonarsource have a look at this PR: https://github.com/SonarSource/sonar-scanner-msbuild/pull/1241/files

it wasn't merged to avoid dropping a .NET Framework-only test. However, IMO you can change just `Tests/SonarScanner.MSBuild.Shim.Test/SonarScanner.MSBuild.Shim.Test.csproj` to `net6` in order to have coverage for the conditional compilation logic

Reply:
I just saw the comment above my latest one (just refeshed). I'll follow the suggestion there.

Reply:
After offline discussion: The `SonarScanner.MSBuild.Shim` is targeting a netstandard version and .NET Framework 4.6

> `<TargetFrameworks>$(ScannerNetStandardVersion);$(ScannerNetFxVersion)</TargetFrameworks>`

This means that the directive  `#if NET` is ignored, and the code in it is not compiled.

Thus, we decided to leave the code with an explanatory comment.